pydatalab/pydatalab/apps/raman/blocks.py

```diff
@@ -30,8 +30,10 @@ def load(self, location: str | Path) -> Tuple[pd.DataFrame, List[str]]:
         ext = os.path.splitext(location)[-1].lower()
 
         if ext == ".txt":
-            df = pd.read_csv(location, sep=r"\s+")
-            df = df.rename(columns={"#Wave": "wavenumber", "#Intensity": "intensity"})
+            try:
```

Comment:
It would be nice to have some additional expected features of the Raman txt format specifically, as at the moment you could upload any two columns of numbers in a txt file and they would be successfully plotted as Raman data.

Looking at the example data, we could also parse the headers and check for various keys, e.g.,

```python
vendor = None # try to detect vendor first
header = []
with open(location, "r") as f:
    while header_line := f.readline().startswith("#"):
        header.append(header_line)
    
    if "#Wave" in header[0] and "#Intensity"  in header[0]:
       vendor = "renishaw"

    # otherwise try to parse the Labstep header
    else:
        metadata = {key: value for key, value in line.split("=") for line in header}
        if metadata.get("AxisType[0]") == "Intens" and metadata.get("AxisType[1]") == "Spectr":
            vendor = "labstep"
            self.data["metadata"] = metadata

    if not vendor:
        raise RuntimeError("Could not detect Raman data vendor -- this file type is not supported by this block.")
            
```

Let me know if you want to have a go at this, otherwise I can do it!

Reply:
That's fair, will give this a go, should be straightforward